cpmpy/transformations/linearize.py

```diff
@@ -128,14 +128,39 @@ def linearize_constraint(lst_of_expr, supported={"sum","wsum"}, reified=False, c
                             continue
                         elif is_false_cst(lin):
                             indicator_constraints=[] # do not add any constraints
-                            newlist+=linearize_constraint([~cond], supported=supported, csemap=csemap) # post linear version of unary constraint
+                            newlist+=linearize_constraint([~cond], supported=supported, csemap=csemap, reified=reified) # post linear version of unary constraint
                             break # do not need to add other
-                        else:
+                        elif "->" in supported and not reified:
                             indicator_constraints.append(cond.implies(lin)) # Add indicator constraint
+                        else: # Nested -> or implication constraints are not supported
```

Comment:
very minor thing, but this comment threw me off (I thought it'd say something about the case we're in). I'd prefer if the comments was moved to the block that checks this support, or if the asserts themselves just tell us this in their message.

Reply:
on the same thread:

- so the assert below should have a message why we expect this comparison, just like the on further below.
- shouldn't we raise the specific excpetion about this, like e.g. TransfNotImplemented ?

Reply:
I'll update the comment to make it a bit more clear indeed.
Regarding the asserts: to me this is just a sanity check, and it should always go through if the rest of **our** code is correct. I.e., the user normally can put in any expression, without triggering this assert